src/preact.d.ts

```diff
@@ -98,18 +98,18 @@ declare namespace preact {
 
 		forceUpdate(callback?: () => void): void;
 
-		abstract render(props?: RenderableProps<P>, state?: Readonly<S>, context?: any): JSX.Element | null;
+		abstract render(props?: RenderableProps<P>, state?: Readonly<S>, context?: any): ComponentChild;
 	}
 
 	function h<P>(
 		node: ComponentFactory<P>,
 		params: Attributes & P | null,
-		...children: (ComponentChild | ComponentChildren)[]
```

Comment:
ComponentChildren = ComponentChild[]

Reply:
Actually preact also accepts an array of ComponentChildren as a valid argument to `h()`
[example](https://preactjs.com/repl?code=import%20%7B%20h%2C%20render%2C%20Component%20%7D%20from%20%27preact%27%3B%0A%0Aconst%20App%20%3D%20()%20%3D%3E%20%7B%0A%09const%20children%20%3D%20%5B%0A%09%09%3Cp%3Echild%201%3C%2Fp%3E%2C%0A%09%09%3Cp%3Echild%202%3C%2Fp%3E%2C%0A%09%09%3Cp%3Echild%203%3C%2Fp%3E%0A%09%5D%3B%0A%09return%20h(%27div%27%2C%20%7B%0A%09%09style%3A%20%27border%3A%201px%20solid%20%23000%3B%27%0A%09%09%7D%2C%0A%09%09children%0A%09)%3B%0A%7D%3B%0Arender(%20%3CApp%20%2F%3E%2C%20document.body)%3B)

Reply:
But If I understand correctly your example your are passing a `ComponentChild[]` (which is an alias for ComponentChildren), not a `ComponentChildren[]` (that would mean `ComponentChild[][]`).

Reply:
I'm indeed passing a `ComponentChildren[]` because `children` is a rest parameter.

Take this example and feed it the typescript compiler:
```typescript
type SomeArrayType = string[];
function test(...test: SomeArrayType) {}
test(['a', 'b', 'c']);
```
e.g. [here](https://www.typescriptlang.org/play/#src=type%20SomeArrayType%20%3D%20string%5B%5D%3B%0D%0Afunction%20test(...test%3A%20SomeArrayType)%20%7B%7D%0D%0Atest(%5B'a'%2C%20'b'%2C%20'c'%5D)%3B) in the typescript playground.

And you'll get the error/warning: `Argument of type 'string[]' is not assignable to parameter of type 'string'.`

If you change the function signature to `(...test: SomeArrayType[])` the error/warning will disapear.

Reply:
Huh, it appears Preact does support `ComponentChild[][]` and flattens it to a single list of children: [example](https://preactjs.com/repl?code=import%20%7B%20h%2C%20render%2C%20Component%20%7D%20from%20%27preact%27%3B%0A%0Aconst%20App%20%3D%20()%20%3D%3E%20%7B%0A%09const%20children%20%3D%20%5B%0A%09%09%5B%3Cp%3Echild%201%3C%2Fp%3E%2C%20%3Cp%3Echild%202%3C%2Fp%3E%5D%2C%0A%09%09%5B%3Cp%3Echild%203%3C%2Fp%3E%2C%20%3Cp%3Echild%204%3C%2Fp%3E%2C%20%3Cp%3Echild%205%3C%2Fp%3E%5D%2C%0A%09%09%5B%3Cp%3Echild%206%3C%2Fp%3E%5D%0A%09%5D%3B%0A%09return%20h(%27div%27%2C%20%7B%0A%09%09style%3A%20%27border%3A%201px%20solid%20%23000%3B%27%0A%09%09%7D%2C%0A%09%09children%0A%09)%3B%0A%7D%3B%0Arender(%20%3CApp%20%2F%3E%2C%20document.body)%3B)

Reply:
ACK! Removed the corresponding commit